### ISSUE TITLE
[no-unescaped-entities] single quote

### ISSUE THREAD

**Issue by tleunen** [yannickcr/eslint-plugin-react#894]
I'm having the `no-unescaped-entities` thrown when a single quote is used.

Example
```jsx
<div>I'm not ok</div>
```

**Comment by ljharb** [comment]
That's correct. You should either:
a) use `{"I'm not ok"}` to make it explicit, or
b) use `‘` (a curly quote) because straight quotes are typographically incorrect.

**Comment by tleunen** [comment]
hmm I see. But writing with curly quotes is quite painful :)
Thanks for the quick answer!

**Comment by ljharb** [comment]
@tleunen on a mac, `⌥-]` or `⌥-}`, on windows, alt 0145 or alt 0146 :-p

**Comment by julienw** [comment]
>  b) use `‘` (a curly quote) because straight quotes are typographically incorrect.

I think you wanted to mention `’` instead of `‘` :)

**Comment by duane** [comment]
Naturally, there are other uses for `'` than contractions and right single quotes. It's disappointing to see a typographic defense of all things for restricting valid input.

**Comment by duane** [comment]
Notably, ASCII 39 (`'`) is primarily apostrophe, aka `&apos;`. This is how most fonts render the glyph; the only subculture that uses this as a right quote are japanese fonts that render it as a right single quotation mark. Furthermore, using single quotation marks in prose is vanishingly rare these days, whereas english contractions and possessives are as common as ever. You're hamstringing prose to support the typographer for no demonstrative benefit in the general case.

**Comment by julienw** [comment]
You can disable the rule if you don't like it.

I'm finding it useful, especially because github highlighting engine goes crazy with an alone apostrophe.

**Comment by ljharb** [comment]
@duane english contractions and possessives should only be curly quotes too. The benefit is that curly quotes can't enclose HTML attribute values, which means the presence of a curly quote is never a bug, but the presence of a straight quote might be one.

**Comment by duane** [comment]
@ljharb 

> @duane english contractions and possessives should only be curly quotes too.

This is incorrect; it's an apostrophe, and it is wildly different than a quotation mark.

**Comment by julienw** [comment]
There are a lot of symbols that look alike. The [Wikipedia page](https://en.wikipedia.org/wiki/Apostrophe) is very clear.
You can use `’` as an apostrophe. You're right, a quotation mark `’` is not correct. Even if that looks alike in most fonts.

**Comment by ljharb** [comment]
@duane technically `'` is not an apostrophe; `‘` is. `'` is only used for notating "minutes" in degree-minute-second notation.

**Comment by duane** [comment]
@ljharb, @julienw, check the ascii spec. `'` is an apostrophe.

**Comment by ljharb** [comment]
I understand that's how computers describe it; however, typography, not the ASCII spec, determines what a character is, and what should be used.

**Comment by jquense** [comment]
this issue is so weird. Why is the react eslint plugin making assertions about my typography, is it going to flag `...` vs `…` as well in a future version?

**Comment by taion** [comment]
See http://www.unicode.org/Public/10.0.0/ucd/NamesList.txt. The Unicode spec explicitly glosses U+2019 as the "preferred character to use for apostrophe":

```
2019	RIGHT SINGLE QUOTATION MARK
	= single comma quotation mark
	* this is the preferred character to use for apostrophe
	x (apostrophe - 0027)
	x (modifier letter apostrophe - 02BC)
	x (heavy single comma quotation mark ornament - 275C)
```

**Comment by taion** [comment]
`'` (U+0027), the apostrophe, is also not the right character to use for minute notation, if we want to be specific. That's actually `′` (U+2032), the prime character.

I`m not sure there are any cases where `'` (U+0027) is the preferred character, though obviously it⁗s convenient and easy to use in many cases.

I think it's quite fair to discourage the use of that character via lint rules, though.

**Comment by jimrandomh** [comment]
For the benefit of other people who come across this: Put `"react/no-unescaped-entities": 0` in the `rules` section of `.eslintrc`. This warning is enforcing a typography geek's pet peeve, and the problem is with the warning, not with your code.

**Comment by ljharb** [comment]
It’s absolutely a problem with your code. That you may not care about correctness doesn’t mean that incorrectness is acceptable to endorse.

**Comment by mmacheerpuppy** [comment]
@jimrandomh Legend.

**Comment by igimanaloto** [comment]
You could also just disable quotes...
```
"react/no-unescaped-entities": [
    "error",
    {
      forbid: [">", "}"],
    },
  ],
```

Had to do it that way, because I still care about <div>>></div> :)

**Comment by gio-built** [comment]
I did `forbid: [">", """, "}"]`

**Comment by robert-orlinski** [comment]
Are you sure that the "forbid" option still works? I am using `["error", { "forbid": [">", "}"] }]` as the `react/no-unescaped-entities` value and still getting errors like:

> Error: `'` can be escaped with `&apos;`, `&lsquo;`, `&#39;`, `&rsquo;`. 

When I set `false` instead of  `["error", { "forbid": [">", "}"] }]` ESLint gets quiet.

**Comment by distinctdan** [comment]
Yeah this is dumb. Single quotes are perfectly valid in HTML, and are the correct way to type english. It's ridiculous to think that any english speaker is going to go look up a bunch of special unicode characters just to type an apostrophe.

**Comment by ljharb** [comment]
@distinctdan no, it is typographically incorrect in english to use straight quotes, sorry.

**Comment by distinctdan** [comment]
@ljharb Regardless of your feelings on quotes, the fact remains that they're valid in HTML and therefore shouldn't be escaped. Also, even though you say straight quotes shouldn't be used, I see in your previous comments that you're using straight quotes for apostrophes. Could this be because it's inconvenient to go lookup special unicode characters like I said?

**Comment by ljharb** [comment]
Lots of things are valid in HTML.

It's because comments are casual, not formal/professional writing like you'd expect on a webpage.

**Comment by julienw** [comment]
The documentation has all the rationale, and gives you information about how to change the set of checked characters. You should go and read it instead of insulting folks providing tools for free on their free time. Here is the documentation: https://github.com/jsx-eslint/eslint-plugin-react/blob/master/docs/rules/no-unescaped-entities.md

**Comment by ljharb** [comment]
Downvote ratio? This plugin is downloaded almost 28 million times a week, and there's less than 100 downvotes here.

If your english strings aren't going through a translation system (which this rule wouldn't be warning about), then you can configure that system to put the proper quotes in, and have straight quotes in your source. If you're not doing i18n, then I'm not sure readability is a high priority for you.

**Comment by ljharb** [comment]
The world already [does](https://www.reddit.com/r/typography/comments/bsndjl/curly_vs_straight_quotation_marks/). This rule is merely enforcing it.

If you don't like it, don't use it, but your personal annoyance or convenience doesn't have any impact on what is correct. This is my project, so it's the place to do whatever I decide to do - welcome to open source.

**Comment by ljharb** [comment]
Comments aren't formal or professional writing like website copy is. There is a huge difference there.

Please do move on, thanks, this won't be changing.
